src/app/core/file/file.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@angular/core";
+import {
+  HttpClient,
+  HttpEventType,
+  HttpProgressEvent,
+  HttpResponse,
+} from "@angular/common/http";
+import { AppSettings } from "../app-config/app-settings";
+import { catchError, concatMap, filter, map } from "rxjs/operators";
+import { Observable } from "rxjs";
+import { MatDialog } from "@angular/material/dialog";
+import { DownloadProgressComponent } from "./download-progress/download-progress.component";
+
+@Injectable()
+export class FileService {
+  private attachmentsUrl = `${AppSettings.DB_PROXY_PREFIX}/${AppSettings.DB_NAME}-attachments`;
+
+  constructor(private http: HttpClient, private dialog: MatDialog) {}
+
+  uploadFile(
+    file: File,
+    entityId: string,
+    property: string,
+    blob = new Blob([file])
```

Comment:
when would file and blob be independent from each other?

Reply:
I used it in the demo mode when I didn't use the http module for requesting the files. This can be removed later.